python/cuml/metrics/_ranking.py

```diff
@@ -61,70 +137,83 @@ def roc_auc_score(y_true, y_score):
         input_to_cuml_array(y_score, check_dtype=[np.int32, np.int64,
                                                   np.float32, np.float64],
                             check_rows=n_rows, check_cols=n_cols)
-
     return _binary_roc_auc_score(y_true, y_score)
 
 
+def _binary_clf_curve(y_true, y_score):
+
+    if y_true.dtype.kind == 'f' and np.any(y_true != y_true.astype(int)):
+        raise ValueError("Continuous format of y_true  "
+                         "is not supported.")
+
+    ids = cp.argsort(-y_score)
+    sorted_score = y_score[ids]
+
+    ones = y_true[ids].astype('float32')  # for calculating true positives
```

Comment:
why float32?

Reply:
I just wanted to convert it to `float` since later on it is used in the RawKernel which expects `float` data type. `float64` should also be fine.

Reply:
I think float then is probably fine. Technically we could use https://github.com/rapidsai/cuml/blob/489a7d80908271245152d7d0be7a32f4faf68928/python/cuml/common/kernel_utils.py#L55 but probably not worth it here, `float` shouldn't pose any problems, correct?

Reply:
I just tried several variations and it is really surprising:
```
# trial 1
ones = y_true[ids]

# trial 2
ones = y_true[ids].astype('float')

#trial 3
ones = y_true[ids].astype('float64')
```
All of them have assertion error due to mismatched results against sklearn.
Only `ones = y_true[ids].astype('float32')` works. Please note that in each trial I not only changed this line but every line with `astype('float32')`.

Reply:
@dantegd Please let me know if this is acceptable. And if there are other changes to be made. Thank you!